asn1/PreSharedKeyV2.asn

```diff
@@ -0,0 +1,46 @@
+PreSharedKeyV2
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt96,
+    UInt128,
+    VarBytes
+    FROM GenericTypes
+;
+
+Psk2EncryptionEnvelope ::= SEQUENCE {
+    -- Random nonce
+    initializationVector UInt96,
+
+    -- Authentication tag (output of AES-GCM encryption)
+    authenticationTag UInt128,
+
+    -- Encrypted PSK Packet
+    -- Note: This is NOT encoded as a variable length (length-prefixed)
+    -- octet string. The size is constrained by the length of the data
+    -- field in the InterledgerPrepare, InterledgerFulfill, or InterledgerReject
+    -- packet containing this Psk2EncryptionEnvelope
+    cipherText OCTET STRING (SIZE (0..32739))
+}
+
+Psk2Packet ::= SEQUENCE {
+    -- Type ID
+    type UInt8,
+
+    -- ID used to correlate requests and responses
+    requestId UInt32,
+
+    -- In PSK2 Requests, this indicates the minimum the receiver should accept
+    -- In PSK2 Responses and Errors, it is the amount that the receiver got
+    requestAmount UInt64,
+
+    -- Application data carried in the encrypted PSK Request/Response/Error
+    data VarBytes
+}
```

Comment:
You are missing the Extensions byte and the junk data.
A sequence has no overall length indicator so you need these to be able to parse this.

The envelope is no use because it wraps encrypted data (unless there is a guarantee that the encrypted data and decrypted data will be the same length). Even if you did figure out the total length from the envelope this would still mean a lot of custom decoding logic and it would be impossible to differentiate between the extensions and the junk data.

Reply:
Those are just the rest of the bytes, which you should ignore anyway. If you don't understand the extensions there's no need to parse them, because unlike in ILP packets you don't need to pass them on

Reply:
In that case I'd remove the mention of extensions completely. We don't actually support extensions if you're expected to just throw away any data you don't expect.

In future we can add new fields and retain the current packet numbers without breaking existing implementations because they will simply throw away the new fields along with the junk data.

The exception would be if we tried to add optional fields which cause a bitmap to be added at the beginning so we'd simply be prevented from adding new optional fields.

Reply:
Removed in https://github.com/interledger/rfcs/pull/384/commits/fd7b42dcf022c0e1be9b34de8f90906799a5dd3b